message/validation/timeValidator.go

```diff
@@ -2,31 +2,29 @@ package validation
 
 import (
 	"errors"
-	"strings"
 	"time"
 )
 
 var (
 	ErrFutureDate  = errors.New("date is too far in the future")
 	ErrPastDate    = errors.New("date is too far in the past")
-	ErrNilTimeFunc = errors.New("currenttime function has not been set")
+	ErrNilTimeFunc = errors.New("current-time function has not been set")
 )
 
 // TimeValidation sees if a given time is within the time frame it is set to validate
 type TimeValidation interface {
-	IsTimeValid(time.Time) (bool, error)
-	CurrentTime() time.Time
+	ValidTime(time.Time) (bool, error)
```

Comment:
I don't think you need `Time` in the function name, but I don't feel that strongly about this.  If you like it, then keep it. 👍